vortex-scan/src/range_scan.rs

```diff
@@ -110,8 +113,15 @@ impl RangeScanner {
                     // If the mask is empty, then we're done.
                     self.state =
                         State::Ready(Canonical::empty(self.scan.result_dtype())?.into_array());
-                } else {
+                } else if self.remaining_filter_conjuncts.is_empty() {
                     self.state = State::Project((mask, self.scan.projection().clone()))
+                } else {
+                    self.mask = mask;
```

Comment:
Can this just be done on line 110?

Reply:
Not without a copy on line 117, since we want to push pass the mask there.